.github/workflows/protect-released-changelog.yml

```diff
@@ -0,0 +1,54 @@
+# This action against that any PR targeting the main branch touches released
+# sections in CHANGELOG file. If change to released CHANGELOG is required, like
+# doing a release, add the \"Unlock Released Changelog\" label to disable this action.
+
+name: Protect released changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches:
+      - main
+jobs:
+  protect-released-changelog:
+    runs-on: ubuntu-latest
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'Unlock Released Changelog')}}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check the released changelog
+        run: |
+          FILE="CHANGELOG.md"
+          TEMP_DIR=$(mktemp -d)
+          echo "Temp folder: $TEMP_DIR"
+
+          # Only the latest commit of the feature branch is available
+          # automatically. To diff with the base branch, we need to
+          # fetch that too (and we only need its latest commit).
+          git fetch origin ${{ github.base_ref }} --depth=1
```

Comment:
The actions/checkout action has a `fetch-depth` option. Could we use it rather than doing things on our own?

Reply:
This command, `git fetch origin ${{ github.base_ref }} --depth=1`, is used to fetch the base branch (main branch). The next command would output the CHANGELOG.md on the base branch to another location for comparison.

I think this is not related to actions/checkout action, which checkout the entire code base to PR.